pkg/cmd/roachprod/install/cluster_synced.go

```diff
@@ -366,22 +366,39 @@ if [ ! -f "{{.Store}}/CURRENT" ]; then
 fi
 {{- end}}
 
-lastpid=""
+# Init with -1 so that when cockroach is initially dead, we print
+# a dead event for it.
+lastpid=-1
 while :; do
 {{ if .Local }}
   pid=$(lsof -i :{{.Port}} -sTCP:LISTEN | awk '!/COMMAND/ {print $2}')
 	pid=${pid:-0} # default to 0
-	status=unknown
+	status="unknown"
 {{- else }}
+  # When CRDB is not running, this is zero.
 	pid=$(systemctl show cockroach --property MainPID --value)
 	status=$(systemctl show cockroach --property ExecMainStatus --value)
 {{- end }}
 
+  if [[ "${lastpid}" == -1 && "${pid}" != 0 ]]; then
+    # On the first iteration through the loop, if the process is running,
+    # don't register a PID change (which would trigger an erroneous dead
```

Comment:
This reads a bit odd, because this _does_ register as a PID change (pid != 0 thus pid != lastpid), but it doesn't register as a dead event.

Reply:
Yeah - I agree. I'm heading out for tonight though and would like the tests to *not* fail again in the upcoming nightly, so going to merge this for now.